chapter_01/exercise_1_12/copy_io_nl.c

```diff
@@ -1,20 +1,27 @@
 #include <stdio.h>
 
+#define OUT 0
+#define IN 1
+
 int main(void)
 {
   char c;
+  int state;

```

Comment:
I am thinking that the current approach can be improved even more (this could be done through another PR). For example, the **state** variable only can be true or false, so it might be nice to be a **boolean** type, but this doesn't come natively in C, so to use something like that it's necessary to build one, maybe a very elegant way might be with a simple **enum**. However, at this point in the book that might be considered advanced, so probably the optimization can be done in a more primitive way. I am thinking that using a **char** for the state might be better because a char will occupy **less space** in memory.

NOTE: It might be interesting to explore something like **unsigned char**.